test/test_any_utils.py

```diff
@@ -6,8 +6,8 @@ def test_key_val_replacement_works(host):
     addOrEditKeyValPair "./testoutput" "KEY_TWO" "value2"
     addOrEditKeyValPair "./testoutput" "KEY_ONE" "value3"
     addOrEditKeyValPair "./testoutput" "KEY_FOUR" "value4"
-    addOrEditKeyValPair "./testoutput" "KEY_FIVE_NO_VALUE"
-    addOrEditKeyValPair "./testoutput" "KEY_FIVE_NO_VALUE"
+    addKey "./testoutput" "KEY_FIVE_NO_VALUE"
+    addKey "./testoutput" "KEY_FIVE_NO_VALUE"
```

Comment:
I'm not sure what you want to do here. Why do you add it twice?

It this is what you really want to check for, adjust the check in L15 to

```
expected_stdout = 'KEY_ONE=value3\nKEY_TWO=value2\nKEY_FOUR=value4\nKEY_FIVE_NO_VALUE\nKEY_FIVE_NO_VALUE\n'
``

Reply:
Ah, no. Thinking on it... AddKey is missing some logic to check for dupes before adding

Reply:
Right. Good Point!

Reply:
I was wrong... we do have a check on L54
````
if ! grep -q "^${key}" "${file}"; then
`````
But still I think it makes not much sense  to add it twice.

Reply:
Ah. Then - we are ok.

The original logic of the test was to prove that adding something that already existed would not duplicate. Maybe we can create a separate test to make this clearer?

Reply:
This is a good idea. 
We have separted the functions and as you wrote in `utils.sh` ;-)
> New functions must have a test added for them in test/test_any_utils.py